pkg/ccl/streamingccl/streamingest/BUILD.bazel

```diff
@@ -50,11 +50,13 @@ go_library(
         "//pkg/sql/pgwire/pgcode",
```

Comment:
My read of the linked issue and the current commit msg is that the `REPLICATION` privilege is specifically to manage the src cluster in a cross-cluster replication job, while the `CONTROLREPLICATION` privilege _seems_ more general and generally applies to cmds executed on the src cluster.

- Is my summarization correct? If so, why not make the privilege names "REPLICATION_SOURCE" and "REPLICATION_DESTINATION"? 
- If there's already been discussion around these names, could you flesh out the commit msg to summarize that discussion?

Reply:
> Is my summarization correct?

`REPLICATION` is the privlege needed by the user to interact with the internal functions that return actual tenant data to the caller.  End-user C2C commands are tied to CONTROLREPLICATION.

>  If so, why not make the privilege names "REPLICATION_SOURCE" and "REPLICATION_DESTINATION"?
> If there's already been discussion around these names, could you flesh out the commit msg to summarize that discussion?

There wasn't a large discussion.  I got to the current break down via the following:

1) Because the internal functions essentially allow you to stream out the entire database, we wanted that to be different from the permission that let's you run `SHOW TENANTS`.
2) PostgreSQL has a role option called REPLICATION that allows the user to create a physical or logical replication stream, so it semantically is similar to what REPLICATION allows here.
3) `CONTROLREPLICATION` felt similar to `CONTROLJOB` and `CONTROLCHANGEFEED` so I used that for consistency when this was still role options.

Reply:
I tend to agree with Steven's reasoning here.

> why not make the privilege names "REPLICATION_SOURCE" and "REPLICATION_DESTINATION"?

Because _in general_  privileges are not meant to be symmetrical with the internal architecture of a feature. Instead, privileges should reflect security perimeters. In this case, they relate to:

- whether or not the user has the ability to exfiltrate a tenant's data, regardless of where it would be exfiltrated to (REPLICATION).
- whether or not the user has the ability to interfere with another user's orchestration / control plane actions (CONTROLXXX)

There's theoretically space for a conversation about what are the security perimeters the customer would really care about. It's mildly possible that both of the above could be served by a single privilege. We should really talk to our customers about that. But what is being done here is fine, as it reflects the separation we've made for other CCL features (eg BACKUP, EXPORT etc).

Reply:
A couple thoughts, solely based on my read of the 23.1.aplha docs:
- It seems the CONTROLXX is the [same style](https://www.cockroachlabs.com/docs/v23.1/create-user#role-options) as our legacy Role Option based authentication scheme for Changefeeds and other jobs that we're [deprecating](https://www.cockroachlabs.com/docs/dev/security-reference/authorization.html). I'm hesitant to add a new  _privilege_ that looks like an old _role option_. I think it is confusing.
- All current supported [privileges](https://www.cockroachlabs.com/docs/dev/security-reference/authorization.html#supported-privileges) seem self explanatory from their names. I find it hard to infer what the REPLICATION and CONTROLREPLICATION privileges really mean from the name. If i were a customer, I could see myself constantly reaching for docs to figure out the difference between these two things, which I'd find mildly annoying.

What is the cost of starting with just one privilege, `REPLICATION`, for everything?

Reply:
> If i were a customer, I could see myself constantly reaching for docs to figure out the difference between these two things, which I'd find mildly annoying.

Assume that we need at least two permissions given my reason above.  Do you have a suggestion for the a name for the second permission?

Reply:
Note that I'm sympathetic to your argument. I think it will currently take careful documentation and people will still be confused about who needs to be granted what on which cluster. So, if we can't decide on a good name, I suppose I could live with it all being under REPLICATION, even if it is a much wider permission than I think most users would need to be granted.

Reply:
Yeah, you're reasons make sense for wanting two different permissions.

My first instinct would be privilege names along the lines of:
Instead of REPLICATION:
- REPLICATION_PRODUCE
- REPLICATION_EXFILTRATE
- REPLICATION_EXPORT

Instead of CONTROLREPLICATION:
- REPLICATION_CONSUME
- REPLICATION_INGEST
- REPLICATION_CONTROL

Reply:
What about `MANAGETENANT`? cc @knz

Reply:
@msbutler @knz I'd like to move forward with this. Unless y'all object, I'll update this to MANAGETENANT.  If we don't like that name, I'd recommend we move forward as-is.

Reply:
MANAGETENANT works for me!

Reply:
would that include all variants of create/drop/alter tenant?

Reply:
@knz I can expand it to cover the other variants of those commands, which I think makes sense if we go with MANAGETENANT as the name.  

But, I'm honestly not sure how I feel about that with respect to `DROP TENANT` and `ALTER TENANT ... GRANT CAPABILITY` especially as the capability system grows to encompass more operations.  As in, say you have an operator that you've given MANAGETENANT in the system tenant and which has ADMIN inside a given tenant.  It could be surprising that such a user could log into the system tenant, increase their tenants capabilities, and then log into their tenant to use those capabilities.